packages/app-check/src/internal-api.ts

```diff
@@ -205,6 +205,89 @@ export async function getToken(
   return interopTokenResult;
 }
 
+/**
+ * This function always resolves.
```

Comment:
If I understand correctly, this is a copy of getToken() without writing the token anywhere? If that's the case can you make some comments indicating that?

Or actually, to reduce code duplication, especially since getToken() needs to be modified a lot for bug fixes and we'd have to make sure to make the fixes in 2 places in this case, maybe it's better to add a 3rd (optional) param to getToken() like `persistToken` (or you can think of a better name) defaulting to true, and wrap all the storage related bits in if blocks that are skipped if it's false?

Reply:
Okay I updated this to run as part of `getToken` instead. I was worried it would add too much complexity but I think I managed to make it not too bad.

Reply:
Is modifying the internal `getToken` necessary to handle the limited-use token case? I understand the code duplication concern, but since the limited-use token API just wants to always obtain a new token without any side effects, doesn't it just need to call through to `provider!.getToken()`? For reference, here's what I'm doing for Android: https://github.com/firebase/firebase-android-sdk/pull/4876/files#diff-b48f738639ba38789c0dcf7a388a16f909140c35a96fc4362c86a01ab10294a9

Reply:
@rosalyntan here's what it looked like on web before integrating with `getToken` --- because of the interop stuff it ends up being more complicated than just calling `provider.getToken()`

https://github.com/firebase/firebase-js-sdk/blob/5dcd94341f5f7b1110b92355760f4a2afc0b606e/packages/app-check/src/internal-api.ts#L213-L289

Reply:
I might not be understanding how web implements the get token flows, but I believe the limited-use token shouldn't need to check for existing token in memory (it always will obtain and return a new token) and since it doesn't affect the stored token, and it shouldn't need to update the interop token result or ever return a dummy token.

Reply:
This implementation doesn't check anything in memory, I think that's a stray comment that's been fixed. I guess you're correct the dummy token part isn't necessary since it seems like this doesn't have any internal consumers and it could just throw if there was an error (the normal use of internal-api getToken() can be consumed both by external users and by other product packages, and in the latter case it can't throw). Sorry, I'm not super familiar with the requirements for this feature, so I don't know which parts were necessary. Does it also not care about debug tokens then?

If it doesn't care about the debug case and it doesn't need a dummy token for interop with other products, then Rosalyn is right, maybe this can be a much shorter separate function, in which case I'm sorry for pointing Sam in the wrong direction.

Reply:
So for the limited-use token, it should behave basically like the 3P getToken(/* forceRefresh=*/ true) _without_ any side-effects -- i.e. it doesn't update the stored/cached token, doesn't reset any proactive refresh timers, doesn't trigger any listeners, etc. In addition, yeah it shouldn't return the dummy token if an error occurs, it should return the error to the caller. @weixifan please correct me if I'm wrong.

It _should_ function properly with the debug provider -- I'm not entirely sure how web handles the debug provider flows, but on iOS and Android, the debug provider is not treated differently than any other provider, so we don't have any special handling for that (i.e. there is no `isDebugMode` casing).